TourMate/TourMate/Controllers/FirebaseTripController.swift

```diff
@@ -32,6 +32,23 @@ struct FirebaseTripController: TripController {
         return (trips, errorMessage)
     }
 
+    func fetchTrip(withTripId tripId: String) async -> (Trip?, String) {
+        guard let user = Auth.auth().currentUser else {
+            return (nil, Constants.messageUserNotLoggedIn)
+        }
+
+        let (adaptedTrip, errorMessage) = await firebasePersistenceManager.fetchItem(id: tripId)
+        guard let adaptedTrip = adaptedTrip as? FirebaseAdaptedTrip else {
+            return (nil, errorMessage)
+        }
+        guard adaptedTrip.attendeesUserIds.contains(user.uid) else {
+            return (nil, errorMessage)
+        }
+
```

Comment:
I think we shouldn't reuse the errorMessage if the persistenceManager manages to fetch something right?
```suggestion
        guard let errorMessage.isEmpty else {
           return (nil, errorMessage)
        }
        guard let adaptedTrip = adaptedTrip as? FirebaseAdaptedTrip else {
            return (nil, "[FirebaseTripController] Error converting FirebaseAdaptedData into FirebaseAdaptedTrip")
        }
        guard adaptedTrip.attendeesUserIds.contains(user.uid) else {
            return (nil, "[FirebaseTripController] Error user \(user.uid) is not an attendee of trip \(tripId)")
        }

```

Reply:
Right!